server/api.js

```diff
@@ -2,100 +2,416 @@ import { Router } from "express";
+			}
+		})
+		.catch((err) => console.error(err.message));
 });
 
-router.post("/register", async (req, res) => {
+//POST a plan for specific user id
+router.post("/users/:user_id/plans", (req, res) => {
+	const { user_id } = req.params;
+	const { description } = req.body;
+	const query =
+		"INSERT INTO plans (user_id, description ) VALUES ($1, $2) RETURNING *";
+	pool
+		.query(query, [user_id, description])
+		.then((result) => {
+			res.json({ success: "New plan has been created!", user: result.rows[0] });
+		})
+		.catch((err) => console.error(err.message));
+});
+
+// GET a plan form specific user with plan id
+router.get("/users/:user_id/plans/:plan_id", (req, res) => {
+	const { user_id, plan_id } = req.params;
+	const query = "SELECT * FROM plans WHERE id=$1 and user_id=$2";
+
+	pool
+		.query(query, [plan_id, user_id])
+		.then((result) => {
+			console.log(result);
+			if (result.rowCount) {
+				res.json(result.rows);
+			} else {
+				res.status(404).json({
+					status: 404,
+					msg: "This user doesn't have a plan with this id",
+				});
+			}
+		})
+		.catch((err) => console.error(err.message));
+});
+
+//UPDATE a plan details with user id
+router.put("/users/:user_id/plans/:plan_id", (req, res) => {
+	const { user_id, plan_id } = req.params;
+	const { description } = req.body;
+	pool
+		.query("SELECT * FROM plans WHERE user_id=$1 and id=$2", [user_id, plan_id])
+		.then((result) => {
+			if (result.rows.length === 0) {
+				return res.status(400).json("We couldn't find user id!");
+			} else {
+				const query = "UPDATE plans SET description=$2 WHERE id=$1 RETURNING *";
+				pool
+					.query(query, [plan_id, description])
+					.then(() =>
+						res.json({ success: "plan has been Updated!", user: result.rows })
+					)
+					.catch((err) => console.error(err.message));
+			}
+		});
+});
+
+// Delete a plan form specific user with plan id <==== have a problem
+router.delete("users/:user_id/plans/:plan_id", async (req, res) => {
+	const { user_id, plan_id } = req.params;
+
+	if (!user_id || !plan_id) {
+		res
+			.status(404)
+			.json({ status: 404, msg: "We couldn't find a plan with this id!" });
+	}
+
 	try {
-		console.log(req.body);
-		const { first_name, last_name, region, role, email, password } = req.body;
-		const newUser = await pool.query(
-			"INSERT INTO users (first_name, last_name, region, role, email, password ) VALUES ($1, $2, $3, $4, $5, $6) RETURNING *",
-			[first_name, last_name, region, role, email, password]
+		const deleteGoal = await pool.query("DELETE FROM goals WHERE plan_id=$1", [
+			plan_id,
+		]);
+
+		const deleteFeedback = await pool.query(
+			"DELETE FROM feedbacks WHERE plan_id=$1",
+			[plan_id]
+		);
+
+		const result = await pool.query(
+			"DELETE FROM plans WHERE plans.id=$1 AND user_id=$2",
+			[plan_id, user_id]
+		);
+
+		res.json({ Success: "Plan has been deleted" });
+	} catch (err) {
+		res.status(500).send("server error");
```

Comment:
It's the first time handling errors this way.
It would be good to update the above endpoints to send 500s as well, but also log errors in the console